dask/array/routines.py

```diff
@@ -2317,8 +2317,10 @@ def coarsen(reduction, x, axes, trim_excess=False, **kwargs):
         + key[1:]: (apply, chunk.coarsen, [reduction, key, axes, trim_excess], kwargs)
         for key in flatten(x.__dask_keys__())
     }
+
     chunks = tuple(
-        tuple(int(bd // axes.get(i, 1)) for bd in bds) for i, bds in enumerate(x.chunks)
+        tuple(int(bd // axes.get(i, 1)) for bd in bds if int(bd // axes.get(i, 1)) > 0)
```

Comment:
nit: Should we encapsulate `int(bd // axes.get(i, 1))` in a more descriptive local function/lambda? Personally, I would find that easier to read.

Reply:
:) yeah I was torn between not adding another line but having a duplicated piece of code, or having a 2-liner but without duplication. Your vote for option 2 helps with the choice.

Reply:
LGTM, I'll merge once CI is green.